addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1142,13 +1142,28 @@ func (r *ClusterBootstrapReconciler) GetDataValueSecretNameFromBootstrapPackage(
 	}
 
 	if cbPkg.ValuesFrom != nil {
+		clusterBootstrapHelper := clusterbootstrapclone.NewHelper(
+			r.context, r.Client, r.aggregatedAPIResourcesClient, r.dynamicClient, r.gvrHelper, r.Log)
 		if cbPkg.ValuesFrom.Inline != nil {
 			packageSecretName := util.GeneratePackageSecretName(cluster.Name, packageRefName)
 			secret := &corev1.Secret{}
 			key := client.ObjectKey{Namespace: cluster.Namespace, Name: packageSecretName}
 			if err = r.Get(r.context, key, secret); err != nil {
-				r.Log.Error(err, "unable to fetch secret for package with inline config", "objectkey", key)
-				return "", err
+				if apierrors.IsNotFound(err) {
+					// secret for package with inline does not exist, we should create one
+					_, err := clusterBootstrapHelper.CreateSecretFromInline(cluster, cbPkg, packageRefName)
```

Comment:
we also need to ensure ownerRef of ClusterBootstrap object on the created secret similar to how it is done for create

Reply:
Fixed